### PR TITLE
[SD-1234] : [Modified] Closing an item should return user to previous de...

### PR DESCRIPTION
**Pull request by sjunaid**
...sk/search view

### CODE REVIEW

File: app/scripts/superdesk/activity/activity.js [hunk 1]
```diff
@@ -264,13 +263,57 @@ define([
...
         /**
+         * Serving for the purpose of setting referrer url via referrer service, also setting url in localStorage. which is utilized to
+         * get last working screen on authoring page if referrer url is unidentified
+         * direct link(i.e from notification pane)
+         *
+         * @param {Object} currentRoute
+         * @param {Object} previousRoute
+         * @returns {string}
+         */
+        this.setReferrer = function(currentRoute, previousRoute) {
```
Comment (petrjasek) [inline app/scripts/superdesk/activity/activity.js]: I think this should be part of `referrer` service

File: app/scripts/superdesk-authoring/authoring.js [hunk 2]
```diff
@@ -359,6 +362,14 @@
 
         $scope.charLimitHit = false;
 
+        if (typeof (referrer.getReferrerUrl()) === 'undefined' || (referrer.getReferrerUrl()) === null){
```
Comment (petrjasek) [inline app/scripts/superdesk-authoring/authoring.js]: all those `if`s could be part of `referrer.getReferrerUrl()` right?

File: app/scripts/superdesk-authoring/authoring.js [hunk 3]
```diff
@@ -430,6 +441,11 @@
                 $scope.item = _.create(item);
                 notify.success(gettext('Item updated.'));
                 startWatch();
+                if (typeof ($scope.referrerUrl) === 'undefined' || $scope.referrerUrl === null){
```
Comment (petrjasek) [inline app/scripts/superdesk-authoring/authoring.js]: this also could be done as part of setting/getting `referrerUrl` and not whenever you use it

Comment (sjunaid) [inline app/scripts/superdesk-authoring/authoring.js]: @petrjasek , that's right making these changes.

File: app/scripts/superdesk/activity/activity.js [hunk 4]
```diff
@@ -267,9 +266,8 @@ define([
 
     module.provider('superdesk', SuperdeskProvider);
 
-    module.service('activityService', ['$location', '$injector', '$q', '$timeout', 'gettext', 'modal',
-    function($location, $injector, $q, $timeout, gettext, modal) {
-
+    module.service('activityService', ['$location', '$injector', '$q', '$timeout', 'gettext', 'modal', 'referrer',
```
Comment (petrjasek) [inline app/scripts/superdesk/activity/activity.js]: seems like `referrer` is not needed here anymore

**Comment by petrjasek** [comment]
:+1: good refactoring

**Pull request merged**